h/services/annotation_moderation.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h import models
+
+
+class AnnotationModerationService(object):
+    def __init__(self, session):
+        self.session = session
+
+    def create(self, annotation):
+        """
+        Create an annotation nipa flag.
+
+        This hides the given annotation from anybody except its author and the
+        group moderators.
+
+        In case the given annotation already has a moderation flag, this method
+        is a no-op.
+
+        :param annotation: The annotation to hide from others.
+        :type annotation: h.models.Annotation
+        """
+
+        query = self.session.query(models.AnnotationModeration) \
+                            .filter_by(annotation=annotation)
+
+        if query.count() > 0:
```

Comment:
Out of interest, is there a clean way to do this in a single query? I’m thinking of something akin to Postres’ [`ON CONFLICT DO NOTHING`](https://www.postgresql.org/docs/9.5/static/sql-insert.html#SQL-ON-CONFLICT) behaviour.

I’m assuming that Postgres’ transactions protect us from the race condition in here, so this is more to satisfy my own curiosity about what SQLAlchemy’s ORM can do.